### ISSUE TITLE
parser: support subsecond resolution with comma (%s,%L)

### ISSUE THREAD

**Issue by divanikus** [fluent/fluent-bit#703]
## Bug Report

I'm trying to parse a simple log file, something like fail2ban.log. It has rather simple time format, but all my attempts to get milliseconds part are failed. Maybe I've missed something?

**To Reproduce**

Consider something like the following:
```
2018-08-03 02:05:59,772 fail2ban.actions        [1665]: NOTICE  [sshd] Unban 58.218.198.172
2018-08-03 02:06:16,195 fail2ban.filter         [1665]: INFO    [sshd] Found 58.218.198.172
2018-08-03 02:06:17,861 fail2ban.filter         [1665]: INFO    [sshd] Found 58.218.198.172
2018-08-03 02:06:20,722 fail2ban.filter         [1665]: INFO    [sshd] Found 58.218.198.172
2018-08-03 02:06:22,527 fail2ban.filter         [1665]: INFO    [sshd] Found 58.218.198.172
```
Parser config:
```
[PARSER]
    Name   fail2ban
    Format regex
    Regex  ^(?<time>\S* \S*)\s+(?<criteria>\S*)\s+\[(?<pid>[0-9]+)\]:\s+(?<priority>\S+)\s+\[(?<service>\S+)\]\s+(?<message>.*?)$
    Time_Key time
    Time_Format %Y-%m-%d %H:%M:%S.%L
```
The result:
```
$ /opt/td-agent-bit/bin/td-agent-bit -i tail -p path=/var/log/fail2ban.log -p parser=fail2ban -R /etc/td-agent-bit/parsers_extra.conf -o stdout
... 
[10311] tail.0: [1533251159.000000000, {"criteria"=>"fail2ban.actions", "pid"=>"1665", "priority"=>"NOTICE", "service"=>"sshd", "message"=>"Unban 58.218.198.172"}]
[10312] tail.0: [1533251176.000000000, {"criteria"=>"fail2ban.filter", "pid"=>"1665", "priority"=>"INFO", "service"=>"sshd", "message"=>"Found 58.218.198.172"}]
[10313] tail.0: [1533251177.000000000, {"criteria"=>"fail2ban.filter", "pid"=>"1665", "priority"=>"INFO", "service"=>"sshd", "message"=>"Found 58.218.198.172"}]
[10314] tail.0: [1533251180.000000000, {"criteria"=>"fail2ban.filter", "pid"=>"1665", "priority"=>"INFO", "service"=>"sshd", "message"=>"Found 58.218.198.172"}]
[10315] tail.0: [1533251182.000000000, {"criteria"=>"fail2ban.filter", "pid"=>"1665", "priority"=>"INFO", "service"=>"sshd", "message"=>"Found 58.218.198.172"}]
```
**Expected behavior**

Timestamps should have non-zero fraction part, for example 1533251159.772000000. I guess it could be somehow related with comma (",") between seconds and milliseconds. I'm using en_US.UTF-8 locale, if any.

**Your Environment**
* Version used: 0.13.6
* Server type and version: Physical server
* Operating System and version: Debian 9
* Filters and plugins: in_tail

**Comment by nokute78** [comment]
We expect `%S.%L` as the milisecond format 
https://github.com/fluent/fluent-bit/blob/master/src/flb_parser.c#L212

As you know, fail2ban log contains"comma" (not "period").
So, fluent-bit can't parse `%S,%L` currently.

**Comment by divanikus** [comment]
So, it's a bug, right? Ruby can parse it with %S,%L btw.

**Comment by nokute78** [comment]
I think it’s spec currently.

fluent-bit uses strptime(3) internally and the API doesn’t support subsecond.

Ruby extends its spec and 
fluent-bit implementation is different.

@edsiper any comments?

**Comment by JavaCS3** [comment]
I think your time format should config like this `Time_Format %Y-%m-%d %H:%M:%S,%L`

**Comment by divanikus** [comment]
@JavaCS3 sure, but then the fluentbit says "Invalid format". You might check yourself.

**Comment by edsiper** [comment]
Given data has a comma instead of a period. Using a comma is not supported. Note that subsecond resolution is a _hack_ since native strptime(3) don't support it. 

Not a bug but an enhancement request.

**Comment by mariomarin** [comment]
[Python applications](https://docs.python.org/3/library/logging.html#logging.Formatter.formatTime) also use comma as their separator for fractional seconds. I am getting this error from Elasticsearch

```
{"took":26,"errors":true,"items":[{"index":{"_index":"microservices-2018.09.17","_type":"flb_7","_id":"d1a23b67-9645-f67d-8a44-b8ffe6e4ee90","status":400,"error":{"type":"mapper_parsing_exception","reason":"failed to parse [time]","caused_by":{"type":"illegal_argument_exception","reason":"Invalid format: \"2018-09-17 13:24:10,244\" is malformed at \" 13:24:10,244\""}}}},{"index":{"_index":"microservices-2018.09.17","_type":"flb_7","_id":"a763bebd-2070-b03e-6e16-bd19e0a46f24","status":400,"error":{"type":"mapper_parsing_exception","reason":"failed to parse [time]","caused_by":{"type":"illegal_argument_exception","reason":"Invalid format: \"2018-09-17 13:24:10,796\" is malformed at \" 13:24:10,796\""}}}},{"index":{"_index":"microservices-2018.09.17","_type":"flb_7","_id":"db88b710-802b-1ad5-0390-025f4f9453ff","status":400,"error":{"type":"mapper_parsing_exception","reason":"failed to parse [time]","caused_by":{"type":"illegal_argument_excepti
```

**Comment by rdxsuper** [comment]
@edsiper Since this issue is still open and also I did not see this '%Y-%m-%d %H:%M:%S,%L' listed in the merged pull request, I assume its not a supported format, yet...

Will there be any support in the future for this time format?

I have a cassandra debug log that contains logs with this time format.

**Comment by edsiper** [comment]
this is already addressed by #782
